src/main/java/cms/gov/madie/measure/services/VersionService.java

```diff
@@ -252,10 +279,21 @@ private List<TestCase> cloneTestCases(List<TestCase> testCases, List<Group> draf
                                       .build())
                           .toList());
                 }
-                return testCase.toBuilder()
-                    .id(ObjectId.get().toString())
-                    .groupPopulations(updatedTestCaseGroupPopulations)
-                    .build();
+                testCase =
+                    testCase.toBuilder()
+                        .id(ObjectId.get().toString())
+                        .groupPopulations(updatedTestCaseGroupPopulations)
+                        .build();
+                if (isFlagEnabled) {
+                  if (caseNumberExists) {
+                    testCase.setCaseNumber(testCase.getCaseNumber());
+                  } else {
+                    testCase.setCaseNumber(sequenceService.generateSequence(measureId));
+                  }
+                } else {
+                  testCase.setCaseNumber(null);
```

Comment:
Does setting `caseNumber` to `null` differ from not setting it at all? If not, then I think this `else` can be removed.

Reply:
@jkotanchik-SB so the caseNumber will be set before the if flagEnabled, I just wanted to make sure if the flag is not enabled, we don't set any value, then again, there probably wouldn't be any value before the flag is enabled, so maybe this is over kill, I'll remove it (I added this part right before the check in)

Reply:
@jkotanchik-SB removed